### ISSUE TITLE
[directive-selector] / [component-selector] False negative when prefix is undefined and style doesn't matches

### ISSUE THREAD

**Issue by Kinqdos** [angular-eslint/angular-eslint#1864]
The rules [directive-selector] and [component-selector] are false negative if no prefix is defined, but the style doesn't matches.
For example kebab-case style for directives doesn't throws an error. When I add a prefix it does (for the missing prefix AND the wrong style)

```JSON
        rules: {
            "@angular-eslint/directive-selector": [
                "error",
                {
                    type: "attribute",
                    style: "camelCase",
                },
            ],
            "@angular-eslint/component-selector": [
                "error",
                {
                    type: "element",
                    style: "kebab-case",
                },
            ],
       }
```
**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `angular-eslint`    | `18.0.1` |
| `typescript-eslint`        | `8.0.0-alpha.28` |
| `ESLint`                           | `9.4.0` |
| `node`                             | `22.1.0` |

```sh
# Please run `npx ng version` in your project and paste the full output here:
Angular CLI: 18.0.3
Node: 22.1.0
Package Manager: pnpm 9.1.3
OS: darwin arm64

Angular: 18.0.2
... animations, cdk, common, compiler, compiler-cli, core, forms
... language-service, material, platform-browser
... platform-browser-dynamic, router

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1800.3
@angular-devkit/build-angular   18.0.3
@angular-devkit/core            18.0.3
@angular-devkit/schematics      18.0.3
@angular/cli                    18.0.3
@schematics/angular             18.0.3
rxjs                            7.8.1
typescript                      5.4.5
zone.js                         0.14.6
```

- [ x] I have tried restarting my IDE and the issue persists.
- [ x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by highlabs** [comment]
Yes, I can confirm that happens at least since version `15.2.1`.
My project:

| package                       | version |
| ----------------------------- | ------- |
| @angular-eslint/eslint-plugin | 15.2.1  |
| eslint                        | 8.55.0  |


My eslint config:

```
          "@angular-eslint/directive-selector": [
              "error",
              {
                "type": "attribute",
                "style": "camelCase"
              }
            ],
            "@angular-eslint/component-selector": [
              "error",
              {
                "type": "element",
                "style": "kebab-case"
              }
            ],
```

**Comment by JamesHenry** [comment]
Fix released as part of https://www.npmjs.com/package/angular-eslint/v/21.0.0
